### PR TITLE
Agora chat restful updates

### PR DESCRIPTION
**Pull request by jinyuagora**

### CODE REVIEW

File: en-US/markdown/agora-chat/RESTful API Reference/agora_chat_restful_message.md [hunk 1]
```diff
@@ -1028,7 +1057,7 @@ curl -X GET -H 'Accept: application/octet-stream' -H 'Authorization: Bearer {You
 
 ## Download a thumbnail
 
-When uploading an image or video fle, the Chat server automatically creates a thumbnail for the file on the server. The difference between this method and the [downloading a file](#download) is that this method has an extra `thumbnail: true` field in the request header. 
+When uploading an image or video file, the Chat server automatically creates a thumbnail for the file on the server. This method has an extra `thumbnail` field in the request header compared with [downloading a file](#download).
```
Comment (mst3k4ever) [inline en-US/markdown/agora-chat/RESTful API Reference/agora_chat_restful_message.md:1060]: Line 899 says videos do not get an automatic thumbnail? Please double check?

File: en-US/markdown/agora-chat/RESTful API Reference/agora_chat_restful_presence.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Presence
 
-The presence feature enables users to publicly display their online presence status and quickly determine the status of others. Users can also customize their presence status, which adds fun and diversity to real-time chatting.
+The presence feature enables users to publicly display their online presence status and quickly determine the status of others. Users can also customize their presence status, such as busy, away, in a call, which adds fun and diversity to real-time chatting.
```
Comment (mst3k4ever) [inline en-US/markdown/agora-chat/RESTful API Reference/agora_chat_restful_presence.md:3]: Edit: "Users can customize their presence status; using more specific terms such as "busy", "away", or "in a call" can add fun and diversity to real-time chatting."  (maybe use more actually "fun" options? For example, something like "terms such as "busy", "on a coffee break", or "currently spacing out"")

File: en-US/markdown/agora-chat/RESTful API Reference/agora_chat_restful_presence.md [hunk 3]
```diff
@@ -195,19 +195,15 @@ For the descriptions of the path parameters, see [Common Parameters](#param).
...
+| `result`    | JSON | Whether the retrieving operation succeeds. If successful, the presence statuses of the users return; otherwise, you can troubleshoot according to the returned reasons. |
 | `uid`       | String     | The unique login account of the user.                |
-| `last_time` | Number     | The Unix timestamp when the user was last online.                                           |
+| `last_time` | Number     | The Unix timestamp when the user was last online, in seconds.                                           |
 | `ext`       | String     | The extension information of the presence status.                |
-| `status`    | JSON Array | The presence statuses on multiple devices of the user.<li>`0`: Offline.<li>`1`: Online.<li>Other strings: The custom presence status defined by the user. |
+| `status`    | JSON | The presence statuses on multiple devices of the user.<ul><li>`0`: Offline.</li><li>`1`: Online.</li><li>Other strings: The custom presence status defined by the user.</li></ul> |
```
Comment (mst3k4ever) [inline en-US/markdown/agora-chat/RESTful API Reference/agora_chat_restful_presence.md:204]: Edit: "Other strings: User-defined custom presence status."

File: en-US/markdown/agora-chat/RESTful API Reference/agora_chat_restful_reaction.md [hunk 4]
```diff
@@ -125,28 +125,28 @@ For each method call, you can retrieve reactions in either one-to-one chats or g
...
 
 #### Path parameter
 
 For the path parameters and the detailed descriptions, see [Common parameters](#param).
 
+#### Query parameter
+
+| Parameter        | Type   | Description                                                       | Required |
+| :---------- | :----- | :------------------------------------------------------------ | :------- |
+| `msgIdList` | Array  |  The ID of the message from which you attempt to retrieve reactions.            | Yes      |
```
Comment (mst3k4ever) [inline en-US/markdown/agora-chat/RESTful API Reference/agora_chat_restful_reaction.md:139]: spacing?

File: en-US/markdown/agora-chat/RESTful API Reference/agora_chat_restful_reaction.md [hunk 5]
```diff
@@ -125,28 +125,28 @@ For each method call, you can retrieve reactions in either one-to-one chats or g
...
+| `msgType`   | String | The chat type.<ul><li>`chat`: One-to-one chats.</li><li>`groupchat`: Group chats.</li></ul>           | Yes     |
```
Comment (mst3k4ever) [inline en-US/markdown/agora-chat/RESTful API Reference/agora_chat_restful_reaction.md:140]: Edit: "The chat type:"

File: en-US/markdown/agora-chat/RESTful API Reference/agora_chat_restful_reaction.md [hunk 6]
```diff
@@ -314,11 +314,11 @@ If the returned HTTP status code is `200`, the request succeeds, and the `data`
 | --- | --- | --- |
 | `requestStatusCode` | String | The status code of this request. `ok` means that the request succeeds. |
 | `reactionId` | String | The reaction ID. |
-| `reaction` | String | The ID of the emoji that is added as the reaction. |
+| `reaction` | String | The emoji ID that is the same as the `message` parameter specified in the request body when [adding a reaction](#create). |
 | `count` | Number | The number of users that have added the reaction. |
-| `state` | String | The state of the this request. |
+| `state` | Bool | Whether the user sending this request has added a reaction to this message.<ul><li>`true`: Yes.</li><li>`false`: No.</li></ul> |
 | `userList` | Array | The list of the users that have added this reaction. It only contains the three user IDs that last used reaction. |
```
Comment (mst3k4ever) [inline en-US/markdown/agora-chat/RESTful API Reference/agora_chat_restful_reaction.md:320]: Edit: "this reaction. Note that this list only contains the last three user IDs to do so." (I think?)

File: en-US/markdown/agora-chat/Reference/agora_chat_receive_webhook.md [hunk 7]
```diff
@@ -241,7 +241,7 @@ When a user sends an audio message in a one-to-one chat, chat group, or chat roo
 | `secret`          | String    | The secret returned after uploading the audio file.  |
 | `file_length`  |  Long  | The size of the audio file in bytes. |
 | `length`    | Int | The duration of the audio file in seconds. |
-| `url`   | String  | The URL of the audio, in the format of `https://{host}/{org_name}/{app_name}/chatfiles/{file_uuid}`, where `file_uuid` is the ID of the audio file. You can fetch `uuid` from the response body after the file is uploaded. |
+| `url`   | String  | The URL of the audio, in the format of `https://{host}/{org_name}/{app_name}/chatfiles/{file_uuid}`, where `file_uuid` is the ID of the audio file. You can fetch `file_uuid` from the response body after the file is uploaded. |
```
Comment (mst3k4ever) [inline en-US/markdown/agora-chat/Reference/agora_chat_receive_webhook.md:244]: Edit: "The URL of the audio file,"

**Review by mst3k4ever — COMMENTED** [review]
Have a wonderful Year of the Rabbit! :)

File: en-US/markdown/agora-chat/RESTful API Reference/agora_chat_restful_contact.md [hunk 8]
```diff
@@ -397,7 +397,9 @@ curl -X GET -H 'Accept: application/json' -H 'Authorization: Bearer {YourAppToke
 
 ## Removing a user from the block list
 
-This method removes the specified user from the block list. After removing a user from the block list, that user resumes their previous relationship with the local user, as a contact or a regular chat user, and can send and receive messages with the local user.
+Removes the specified user from the block list. After removing a user from the block list, that user resumes their previous relationship with the current user, as:
+- a contact who can send and receive messages with the current user, or
+- a stranger who has not been added as contact.
```
[the remark this answers is not in the document]
Comment (jinyuagora) [reply en-US/markdown/agora-chat/RESTful API Reference/agora_chat_restful_contact.md:402]: Emm if they know each other, and were blocked for a reason, they resume contact to each other after the unblock operation I guess?

File: en-US/markdown/agora-chat/RESTful API Reference/agora_chat_restful_message.md (hunk 1 above)
Comment (jinyuagora) [reply en-US/markdown/agora-chat/RESTful API Reference/agora_chat_restful_message.md:1060]: Nice spot. Videos do not get an automatic thumbnail. I will modify the description here.

File: en-US/markdown/agora-chat/RESTful API Reference/agora_chat_restful_contact.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (mst3k4ever) [reply en-US/markdown/agora-chat/RESTful API Reference/agora_chat_restful_contact.md:402]: I'm mostly looking at the word "stranger", which means "someone I don't know". That might not be the case for blocked (or formerly blocked) people.

File: en-US/markdown/agora-chat/RESTful API Reference/agora_chat_restful_message.md [hunk 9]
```diff
@@ -1028,7 +1057,7 @@ curl -X GET -H 'Accept: application/octet-stream' -H 'Authorization: Bearer {You
 
 ## Download a thumbnail
 
-When uploading an image or video fle, the Chat server automatically creates a thumbnail for the file on the server. The difference between this method and the [downloading a file](#download) is that this method has an extra `thumbnail: true` field in the request header. 
+When uploading an image or video file, the Chat server can create a thumbnail for the file. This method has an extra `thumbnail` field in the request header compared with [downloading a file](#download).
```
Comment (haoxiuwen) [inline en-US/markdown/agora-chat/RESTful API Reference/agora_chat_restful_message.md:1060]: When uploading an image or video file, the Chat server can create a thumbnail for the file. This method has an extra `thumbnail: true` field in the request header compared with [downloading a file](#download).

Comment (haoxiuwen) [reply en-US/markdown/agora-chat/RESTful API Reference/agora_chat_restful_message.md:1060]: | `thumbnail` | Bool | 是否下载缩略图：<ul><li> `true`：是，下载缩略图。</li><li> `false`：否，下载原文件。</li></ul> <div class="alert info">若该参数为空，下载原文件。<div> | 否 |

File: en-US/markdown/agora-chat/RESTful API Reference/agora_chat_restful_contact.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (jinyuagora) [reply en-US/markdown/agora-chat/RESTful API Reference/agora_chat_restful_contact.md:402]: I will change "stranger" to "user" then. This avoids ambiguity.

File: en-US/markdown/agora-chat/RESTful API Reference/agora_chat_restful_message.md (hunk 9 above)
Comment (jinyuagora) [reply en-US/markdown/agora-chat/RESTful API Reference/agora_chat_restful_message.md:1060]: Done

**Pull request merged**